public/containers/RefreshAction/index.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from "@elastic/eui";
+import { CoreStart } from "opensearch-dashboards/public";
+import { ServicesContext } from "../../services";
+import { CoreServicesContext } from "../../components/core_services";
+import { aliasBlockedPredicate, dataStreamBlockedPredicate, filterBlockedItems, indexBlockedPredicate } from "../../utils/helpers";
+import { IndexOpBlocksType, INDEX_OP_TARGET_TYPE } from "../../utils/constants";
+import { CatIndex, DataStream } from "../../../server/models/interfaces";
+import { IAlias } from "../../pages/Aliases/interface";
+
+interface RefreshActionModalProps<T> {
+  selectedItems: T[];
+  visible: boolean;
+  onClose: () => void;
+  type: string;
+}
+
+export default function RefreshActionModal<T>(props: RefreshActionModalProps<T>) {
+  const { onClose, visible, selectedItems, type } = props;
+  const services = useContext(ServicesContext);
+  const coreServices = useContext(CoreServicesContext) as CoreStart;
+  const [unBlockedItems, setUnBlockedItems] = useState([] as string[]);
+  const [blockedItems, setBlockedItems] = useState([] as string[]);
+
+  useEffect(() => {
+    if (!!services && visible) {
+      switch (type) {
+        case INDEX_OP_TARGET_TYPE.INDEX:
+          filterBlockedItems<CatIndex>(services, selectedItems as CatIndex[], IndexOpBlocksType.Closed, indexBlockedPredicate).then(
```

Comment:
We should add .catch logic to handle errors(e.g. no permission) in filterBlockedItems function. I think in this case we can degrade to not filtering blocked items, i.e. set selectedItems as nonBlockedItems.